RecoHI/HiJetAlgos/plugins/HiBadParticleFilter.cc

```diff
@@ -0,0 +1,298 @@
+  produces<bool>();
+  produces<reco::PFCandidateCollection>();
+  produces<reco::PFCandidateCollection>("cleaned");
+}
+
+HiBadParticleFilter::~HiBadParticleFilter() {}
+
+//
+// member functions
+//
+
+// ------------ method called on each new Event  ------------
+bool HiBadParticleFilter::filter(edm::StreamID iID, edm::Event& iEvent, const edm::EventSetup& iSetup) const {
+  using namespace std;
+  using namespace edm;
+
+  typedef View<reco::PFCandidate> CandidateView;
+  Handle<CandidateView> pfCandidates;
+  iEvent.getByToken(tokenPFCandidates_, pfCandidates);
+
+  const reco::VertexCollection* recoVertices;
+  edm::Handle<reco::VertexCollection> vertexCollection;
+  iEvent.getByToken(tokenPV_, vertexCollection);
+  recoVertices = vertexCollection.product();
+
+  auto pOutputCandidateCollection = std::make_unique<reco::PFCandidateCollection>();
+  auto pBadCandidateCollection = std::make_unique<reco::PFCandidateCollection>();
+
+  bool foundBadCandidate = false;
+
+  for (unsigned j = 0; j < pfCandidates->size(); ++j) {
+    const reco::PFCandidate& pfCandidate = (*pfCandidates)[j];
+
+    if (abs(pfCandidate.particleId()) == 3)  // muon cleaning
+    {
+      if (pfCandidate.pt() > minMuonPt_) {
+        if (!pfCandidate.muonRef()->isGlobalMuon() || !pfCandidate.muonRef()->isTrackerMuon() ||
+            !pfCandidate.trackRef().isNonnull()) {
+          if (verbose_)
+            std::cout << " bad muon fit " << pfCandidate.pt() << std::endl;
+          cout << " isGlobal " << pfCandidate.muonRef()->isGlobalMuon() << " isTracker "
+               << pfCandidate.muonRef()->isTrackerMuon() << " has track " << pfCandidate.trackRef().isNonnull()
+               << std::endl;
+          foundBadCandidate = true;
+          continue;
+        }
+        reco::TrackRef track = pfCandidate.trackRef();
+
+        if (track->ptError() / track->pt() > minMuonTrackRelPtErr_ || track->pt() < pfCandidate.pt() / 2.) {
+          if (verbose_) {
+            std::cout << " bad muon err " << pfCandidate.pt() << std::endl;
+            std::cout << " rel err = " << track->ptError() / track->pt() << std::endl;
+          }
+          foundBadCandidate = true;
+          continue;
+        }
+
+        if (track->algo() == 13 || track->algo() == 14 || track->originalAlgo() == 14 || track->originalAlgo() == 13 ||
+            track->hitPattern().trackerLayersWithMeasurement() < 7) {
+          float xVtx = (*recoVertices)[0].position().x();
+          float yVtx = (*recoVertices)[0].position().y();
+          float zVtx = (*recoVertices)[0].position().z();
+          float xVtxErr = (*recoVertices)[0].xError();
+          float yVtxErr = (*recoVertices)[0].yError();
+          float zVtxErr = (*recoVertices)[0].zError();
+
+          math::XYZPoint vtx_temp(xVtx, yVtx, zVtx);
+
+          float Dz = track->dz(vtx_temp);
+          float DzError = sqrt(track->dzError() * track->dzError() + zVtxErr * zVtxErr);
```

Comment:
- No reason to compute the `sqrt` if you have to square it again on L144 (same for `DxyError`)
- Lower case initials here, as for other local variables in this class